src/app/api/user/full-name/route.ts

```diff
@@ -0,0 +1,42 @@
+import { response200, response400, response401 } from "@/app/utils";
+import { selectUserById, updateUserFullName } from "@/model/UserModel";
+import getUserServerSide from "@/services/server/UserService";
+import { z } from "zod";
+
+const requestType = z.object({
+  full_name: z.string().trim().min(1),
+});
+
+export async function POST(request: Request) {
+  const user = await getUserServerSide();
+  if (user === undefined) {
+    return response401("User not logged in");
+  }
+  let body = undefined;
+  try {
+    body = await request.json();
+  } catch (err) {
+    console.error(err);
+    return response400("Body needs to be in JSON format");
+  }
```

Comment:
See feedback from the previous route. Then replace with re-usable function here.